bounded/src/Data/Bounded/TypeLevel.hs

```diff
@@ -26,23 +26,21 @@ type IsInRange (nat :: Nat) (min :: Nat) (max :: Nat)
 -- | Constructor for appending a type level string with a normal Type
 data (a :: Symbol) ++: (b :: Type)
 
-type family (++) (a :: Symbol) (b :: k) :: Symbol where
-  a ++ (b :: Symbol) = a `AppendSymbol` b
-
 typeName :: forall (a :: Type). Typeable a => String
 typeName
   = fromMaybe (error "GHC error invalid type name")
   . List.stripPrefix "Proxy * "
   . show
   $ typeOf (Proxy @a)
 
+type Length :: Symbol -> Nat
 type family Length (a :: Symbol) :: Nat where
```

Comment:
Comment (Not a blocker): Now that we can have a top-level type definition for `Length`, the inlined type annotations seem redundant.